Fix CAgg on CAgg different column order

fabriziomello opened the pull request:
Creating a CAgg on a CAgg where the time column is in a different order of the original hypertable was raising the following exception:

`ERROR:  time bucket function must reference a hypertable dimension column`

The problem was during the validation we're initializing internal data structure with the wrong hypertable metadata.

Fixes #5131

RafiaSabih commented on tsl/test/expected/cagg_on_cagg.out:
```diff
@@ -18,22 +18,33 @@
-  temperature NUMERIC
-);
+DROP TABLE IF EXISTS conditions CASCADE;
+psql:include/cagg_on_cagg_setup.sql:14: NOTICE:  table "conditions" does not exist, skipping
```
Is this necessary?

fabriziomello replied:
At this point no but the new file `include/cagg_on_cagg_setup.sql` will be included several times so we'll need to cleanup the hypertable to re-setup the test case

akuzm reviewed (approved):
Weird that codecov complains about the lines that should be covered...

fabriziomello commented:
> Weird that codecov complains about the lines that should be covered...

It is really weird. In another simple fix I've added a failing test and codecov complains exactly the code I added to fix the failing test  :-(
